src/source.ts

```diff
@@ -116,7 +116,8 @@ export default class Source extends React.Component<Props> {
     if (
       geoJsonSource &&
       props.geoJsonSource &&
-      props.geoJsonSource.data !== geoJsonSource.data
+      props.geoJsonSource.data !== geoJsonSource.data &&
```

Comment:
What is the intended functionality for this line? Is this supposed to be a deep equal?